app/services/proofing/resolution/progressive_proofer.rb

```diff
@@ -83,17 +86,21 @@ def proof_with_threatmetrix_if_needed
         ddp_pii[:email] = user_email
         ddp_pii[:request_ip] = request_ip
 
-        timer.time('threatmetrix') do
+        result = timer.time('threatmetrix') do
           lexisnexis_ddp_proofer.proof(ddp_pii)
         end
+        add_sp_cost(:threatmetrix, result.transaction_id)
+        result
```

Comment:
if you feel like getting fancy/tricky with Ruby blocks to avoid a local variable, you could do:
```suggestion
        timer.time('threatmetrix') do
          lexisnexis_ddp_proofer.proof(ddp_pii)
        end.tap do |result|
          add_sp_cost(:threatmetrix, result.transaction_id)
        end
```

Reply:
Ah yeah this is prettier. I never think to use `#tap`.